### PR TITLE
Add smaller version of strtol with fewer dependencies

### PR DESCRIPTION
**Pull request by sbc100**
This also means that other functions that depend on strtol such
as inet_aton can be used without bringing in more code.

I measures the size of an otherwise empty program that only
references strtol and built with -Oz:

Before:
wasm: 2512
js:   11509

After:
wasm: 827
js:   11510

The minor change the test expectations actually brings the output
in line with that glibc produces so I think this might actually
be a bugfix too.

### CODE REVIEW

File: src/library_syscall.js [hunk 1]
```diff
@@ -770,7 +770,7 @@ var SyscallsLibrary = {
   },
   __sys_wait4__proxy: false,
   __sys_wait4: function(pid, wstart, options, rusage) {
-    abort('cannot wait on child processes');
+    return -{{{ cDefine('ENOSYS') }}}; // unsupported feature
```
Comment (kripken) [inline src/library_syscall.js:773]: this looks like a merge issue? this landed on master i believe. there are also deps_info changes here that looks unintended.

File: tests/core/test_strtoll_oct.c [hunk 2]
```diff
@@ -14,15 +15,24 @@ int main() {
...
   long long int l5 = strtoll(end_char, &end_char, 8);
+  assert(l5 == -29);
   long long int l6 = strtoll(end_char, NULL, 8);
+  assert(l6 == 2505);
 
-  printf("%d%d%d%d%d%d\n", l1 == 0, l2 == -29, l3 == 2505, l4 == 0, l5 == -29,
-         l6 == 2505);
+  printf("%lli\n", l1);
+  printf("%lli\n", l2);
+  printf("%lli\n", l4);
```
Comment (kripken) [inline tests/core/test_strtoll_oct.c:34]: `l3` is missing here

File: tests/other/metadce/hello_libcxx_O2_fexceptions.exports [hunk 3]
```diff
@@ -1,4 +1,3 @@
-_ZSt18uncaught_exceptionv
```
Comment (kripken) [inline tests/other/metadce/hello_libcxx_O2_fexceptions.exports:1]: why did this change?

**Comment by kripken** [comment]
Are the before/after numbers flipped? (or maybe I misread it)

How did you derive the code here? Is it written from scratch, or modified from the existing musl one, or otherwise?

File: src/library_syscall.js (hunk 1 above)
Comment (sbc100) [reply src/library_syscall.js]: This is not part of the change..  it seems the github gets confused when the upstream of a given PR is merged into master.  I've reuploaded.

**Comment by sbc100** [comment]
Yes I had before and after flipped.  Fixed.

**Comment by sbc100** [comment]
Added comment to the new code

**Comment by sbc100** [comment]
Should be good to go now.

**Comment by kripken** [comment]
Sorry, I don't think you answered my question from earlier:

> How did you derive the code here? Is it written from scratch, or modified from the existing musl one, or otherwise?

I ask because if it's written from scratch then I'd want to review it line by line very carefully.

File: tests/core/test_strtol.out [hunk 4]
```diff
@@ -208,8 +208,6 @@ strtol("012", 0, 10) = 12
 strtol("0y12", 0, 0) = 0
 strtol("hello", 0, 30) = 14167554
 strtol("hello", 0, 10) = 0
-ERR Invalid argument
 strtol("not-a-number") = 0
-ERR Invalid argument
```
Comment (kripken) [inline tests/core/test_strtol.out:213]: why did this change?

Comment (kripken) [reply tests/core/test_strtol.out:213]: Oh, is this the EINVAL issue mentioned earlier?

**Review by kripken — COMMENTED** [review]
lgtm assuming the code is not new, as it looks generally correct.

If it is new or heavily modified then I can find time to read it more carefully.

File: tests/core/test_strtol.out (hunk 4 above)
Comment (sbc100) [reply tests/core/test_strtol.out:213]: This is the change i mention in the description.     I also added a line about this to the comment.   The new behaviour is slightly different in this one edge case (where the very first char is bigger than base) but matches glibc so I think its actually an improvment.

**Comment by sbc100** [comment]
> Sorry, I don't think you answered my question from earlier:
> 
> > How did you derive the code here? Is it written from scratch, or modified from the existing musl one, or otherwise?
> 
> I ask because if it's written from scratch then I'd want to review it line by line very carefully.

I added a comment to the code describing where this is derived from and how it differs.  Hopefully that is enough.

**Comment by sbc100** [comment]
Landing, since the code is largely borrowed from the existing __intscan and we have fairly good testing I believe.
